### PR TITLE
Orders api

### PR DESCRIPTION
**Pull request by mayankamencherla**
Updated code base with Orders API with auto-capture feature enabled.

### CODE REVIEW

**Comment by captn3m0** [comment]
@mayankamencherla DS_Store files committed.

**Comment by captn3m0** [comment]
@mayankamencherla Please break your commits. Especially the one that adds the entire SDK. Add that commit to master and remove it from this branch, so we can review this easily.

Same as what I did on https://github.com/razorpay/razorpay-prestashop/pull/2

Also, `.DS_Store` files are committed here.

**Comment by mayankamencherla** [comment]
@captn3m0 Removed DS_Store files.

File: catalog/controller/payment/razorpay-sdk/Razorpay.php [hunk 1]
```diff
@@ -0,0 +1,42 @@
+<?php
+
+if(class_exists('Requests') === false){
+    require_once __DIR__.'/libs/Requests-1.6.1/library/Requests.php';
```
Comment (73sl4) [inline catalog/controller/payment/razorpay-sdk/Razorpay.php]: move the sdk to the system directory

File: catalog/controller/payment/razorpay.php [hunk 2]
```diff
@@ -36,74 +46,89 @@ protected function index()
...
+                {   
+                    $payment = $api->payment->fetch($razorpay_payment_id);
+                }
+                else
+                {   
+                    $signature = hash_hmac('sha256', $razorpay_order_id . '|' . $razorpay_payment_id, $key_secret);
 
-            try {
-                $ch = $this->get_curl_handle($razorpay_payment_id, $amount);
+                    if (hash_equals($signature , $razorpay_signature))
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: call $this->hash_equals and decide there

File: catalog/controller/payment/razorpay.php [hunk 3]
```diff
@@ -36,74 +46,89 @@ protected function index()
...
-        curl_setopt($ch, CURLOPT_CAINFO, dirname(__FILE__).'/ca-bundle.crt');
-
-        return $ch;
+        $order = $this->model_checkout_order->getOrder($this->session->data['order_id']);
+        
+        switch($this->payment_action)
+        {
+            case 'authorize':
+                $data = array(
+                  'receipt' => $order_id,
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: we don't need entire data twice, only the payment _capture field changes.

File: catalog/view/theme/default/template/payment/razorpay.tpl [hunk 4]
```diff
@@ -46,8 +49,10 @@
 
 </script>
 <form name="razorpay-form" id="razorpay-form" action="<?php echo $return_url; ?>" method="POST">
-  <input type="hidden" name="razorpay_payment_id" id="razorpay_payment_id" />
   <input type="hidden" name="merchant_order_id" id="merchant_order_id" value="<?php echo $merchant_order_id ?>"/>
+  <input type="hidden" name="razorpay_payment_id" id="razorpay_payment_id" />
+  <input type="hidden" name="razorpay_order_id" id="razorpay_order_id" />
```
Comment (73sl4) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: we should not be submitted order id here

File: catalog/view/theme/default/template/payment/razorpay.tpl [hunk 5]
```diff
@@ -15,8 +15,11 @@
     notes: {
       opencart_order_id: "<?php echo $merchant_order_id; ?>"
     },
+    order_id: "<?php echo $razorpay_order_id; ?>",
     handler: function (transaction) {
-        document.getElementById('razorpay_payment_id').value = transaction.razorpay_payment_id;
+        document.getElementById('razorpay_payment_id').value = transaction.razorpay_payment_id; // orders API not done
+        document.getElementById('razorpay_order_id').value = transaction.razorpay_order_id;
```
Comment (73sl4) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: drop order id from here as well

File: catalog/view/theme/default/template/payment/razorpay.tpl (hunk 4 above)
Comment (mayankamencherla) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: dropped

File: catalog/view/theme/default/template/payment/razorpay.tpl (hunk 5 above)
Comment (mayankamencherla) [inline catalog/view/theme/default/template/payment/razorpay.tpl]: dropped

File: catalog/controller/payment/razorpay.php (hunk 2 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: added

File: catalog/controller/payment/razorpay.php (hunk 3 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: added

File: catalog/controller/payment/razorpay-sdk/Razorpay.php (hunk 1 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay-sdk/Razorpay.php]: added

File: catalog/controller/payment/razorpay.php [hunk 6]
```diff
@@ -16,6 +19,12 @@ protected function index()
 
         $order_info = $this->model_checkout_order->getOrder($this->session->data['order_id']);
 
+        // Orders API with payment autocapture
+        $api = new Api($this->config->get('razorpay_key_id'), $this->config->get('razorpay_key_secret'));
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: Assign key_id and key_secret to variables and use them.

File: catalog/controller/payment/razorpay.php [hunk 7]
```diff
@@ -36,74 +46,76 @@ protected function index()
...
             $success = false;
-            $error = '';
+            $error = "";
+            $captured = false;
+
+            try 
+            {
+                if ($this->payment_action === 'authorize')
+                {   
+                    $payment = $api->payment->fetch($razorpay_payment_id);
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: Remove if block.

File: catalog/controller/payment/razorpay.php [hunk 8]
```diff
@@ -36,74 +46,76 @@ protected function index()
...
+                }
+                else
+                {   
+                    $signature = hash_hmac('sha256', $razorpay_order_id . '|' . $razorpay_payment_id, $key_secret);
 
-            try {
-                $ch = $this->get_curl_handle($razorpay_payment_id, $amount);
+                    if ($this->hash_equals($signature , $razorpay_signature))
+                    {
+                        $captured = true;;
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: `$captured` variable is redundant. Use `$success` directly.

File: catalog/controller/payment/razorpay.php [hunk 9]
```diff
@@ -36,74 +46,76 @@ protected function index()
...
-                $result = curl_exec($ch);
-                $http_status = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+                //Check success response
+                if ($captured)
+                {
+                    $success = true;
+                }
 
-                if ($result === false) {
+                else{
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: brackets in next line.

File: catalog/controller/payment/razorpay.php [hunk 10]
```diff
@@ -36,74 +46,76 @@ protected function index()
...
-                    } else {
-                        $success = false;
-
-                        if (!empty($response_array['error']['code'])) {
-                            $error = $response_array['error']['code'].':'.$response_array['error']['description'];
-                        } else {
-                            $error = 'RAZORPAY_ERROR:Invalid Response <br/>'.$result;
-                        }
-                    }
+                    $error = "PAYMENT_ERROR = Payment failed";
```
Comment (73sl4) [inline catalog/controller/payment/razorpay.php]: Use semicolon with =

Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: using

File: catalog/controller/payment/razorpay.php (hunk 9 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: removed

File: catalog/controller/payment/razorpay.php (hunk 8 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: removed

File: catalog/controller/payment/razorpay.php (hunk 7 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: removed

File: catalog/controller/payment/razorpay.php (hunk 6 above)
Comment (mayankamencherla) [inline catalog/controller/payment/razorpay.php]: added

**Pull request merged**